webapp/src/components/jira_field.jsx

```diff
@@ -75,6 +75,22 @@ export default class JiraField extends React.PureComponent {
 
         if (field.allowedValues && field.allowedValues.length) {
             const options = field.allowedValues.map(this.makeReactSelectValue);
+
+            if (field.schema.system === 'priority') {
+                return (
+                    <ReactSelectSetting
+                        key={field.key}
+                        name={field.key}
+                        label={field.name}
+                        options={options}
+                        required={this.props.obeyRequired && field.required}
+                        onChange={this.handleChange}
+                        isMulti={false}
+                        value={options.filter((option) => option.value === this.props.value.id)}
```

Comment:
Note to reviewers.  I kept with the spirit of the other components and created a unique component for the priority special case.  I am curious about your thoughts regarding the following alternate solution:

Instead of creating an additional component, I could use the if statement to set a variable to `this.props.value.id` or `this.props.value`.  Note this is the only difference between the two components, in setting the value property.

Reply:
@crspeller is the SelectSetting below the check `if (field.allowedValues && field.allowedValues.length)` used for any other field other than priority? I don't think so, since there isn't any other field that has `allowedValues` (other than project and and issueType which are handled previously anyway).

In which case you don't have to make this special case, just use the original below (and maybe make it more specific like you have here).

Then, can you just alter priority's field structure to conform to what we expect. That would mean you could remove the special case you added to `initialState` and `handleFieldChange` in `create_issue.jsx`. Make sense?

Reply:
Ya I think it is used in some other fields. You can also create a custom field that would have allowed values.
This should be generic though special casing this here is not what we want. Just apply your modifications to below.